llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -6053,9 +6053,34 @@ Value *SPIRVToLLVM::transSPIRVImageQueryLodFromInst(SPIRVInstruction *bi, BasicB
   // A sampler descriptor is encoded as {desc,convertingSamplerIdx}. Extract the actual sampler.
   Value *samplerDesc = getBuilder()->CreateExtractValue(imageInfo.samplerDesc, 0);
 
-  // Generate the operation.
+  // Generate the operation for normal image get lod.
   Value *coord = transValue(bii->getOpValue(1), bb->getParent(), bb);
-  return getBuilder()->CreateImageGetLod(imageInfo.dim, imageInfo.flags, imageInfo.imageDesc, samplerDesc, coord);
+  Value *result =
+      getBuilder()->CreateImageGetLod(imageInfo.dim, imageInfo.flags, imageInfo.imageDesc, samplerDesc, coord);
+
+  // Extract sampler descriptor for ycbcr, otherwise the desc would be null.
+  if (!m_convertingSamplers.empty()) {
+    // Refer to transSPIRVImageSampleFromInst for details.
+    Value *convertingSamplerIdx = getBuilder()->CreateExtractValue(imageInfo.samplerDesc, 1);
```

Comment:
This is nearly the same code as in transSPIRVImageSampleFromInst. It should be commoned up. I suggest a helper func that does the select ladder thing, and you pass it a lambda to do the actual operation (CreateImageGetLod in this occurrence, CreateImageSampleConvert in the other occurrence).

Reply:
+1

Reply:
> This is nearly the same code as in transSPIRVImageSampleFromInst. It should be commoned up. I suggest a helper func that does the select ladder thing, and you pass it a lambda to do the actual operation (CreateImageGetLod in this occurrence, CreateImageSampleConvert in the other occurrence).

Right, Implemented.